### PR TITLE
feat: Token description

### LINKED ISSUE #1926: Token: Create - Enter Description
Input box has ability to enter a description.   Not a required field.  Must generate a field name, if none provided.

Generate field name: 
{READ}{WRITE}{RESOURCE_NAME}{RESOURCE_SUBNAME}&nbsp{repeat for all resources}

What are the validation requirement requirements?

### PR DESCRIPTION
**Pull request by blegesse-git**
Closes #1926 

The custom api token overlay now has an input box for the description of token. 
Screen Shot 2021-07-27 at 10 35 50 AM

### CODE REVIEW

File: src/authorizations/components/CustomApiTokenOverlay.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      <Overlay.Body>
+        <Form>
+          <FlexBox
+            alignItems={AlignItems.Center}
+            direction={FlexDirection.Column}
+            margin={ComponentSize.Large}
+          >
+            <Form.Element label="Description">
+              <Input
+                placeholder="your_token_description"
```
Comment (hoorayimhelping) [inline src/authorizations/components/CustomApiTokenOverlay.tsx:43]:
```suggestion
                placeholder="Enter your token's description"
```

looks like Figma had a placeholder. @daviddkkim what do you think?

File: src/authorizations/components/CustomApiTokenOverlay.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  AlignItems,
+  FlexDirection,
+  ComponentSize,
+} from '@influxdata/clockface'
+
+interface OwnProps {
+  onClose: () => void
+}
+
+const CustomApiTokenOverlay: FC<OwnProps> = props => {
```
Comment (hoorayimhelping) [inline src/authorizations/components/CustomApiTokenOverlay.tsx:18]:
```suggestion
export const CustomApiTokenOverlay: FC<OwnProps> = props => {
```

File: src/authorizations/components/CustomApiTokenOverlay.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+              />
+            </Form.Element>
+          </FlexBox>
+        </Form>
+      </Overlay.Body>
+    </Overlay.Container>
+  )
+}
+
+export default CustomApiTokenOverlay
```
Comment (hoorayimhelping) [inline src/authorizations/components/CustomApiTokenOverlay.tsx:56]:
```suggestion
```
I'm trying to get the team away from default exports: https://basarat.gitbook.io/typescript/main-1/defaultisbad

File: src/authorizations/components/redesigned/CustomApiTokenOverlay.tsx [hunk 4]
```diff
@@ -1,22 +1,56 @@
...
+              />
+            </Form.Element>
+          </FlexBox>
+        </Form>
+      </Overlay.Body>
     </Overlay.Container>
   )
 }
+
+export default CustomApiTokenOverlay
```
Comment (hoorayimhelping) [inline src/authorizations/components/redesigned/CustomApiTokenOverlay.tsx:56]: same thing here with default exports

**Review by hoorayimhelping — CHANGES_REQUESTED** [review]

**Comment by ui-ci-bot-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 62 | :yellow_circle:
accessibility | 96 | :green_circle:
seo | 90 | :green_circle:
pwa | null | :hankey:
best-practices | 100 | :green_circle:


[Compare your scores with master branch](https://influxdb.aws.influxdata.io/orgs/844910ece80be8bc/dashboards/073c4b95f1089000?lower=now%28%29%20-%202d&vars%5Blighthouse_score_branch%5D=2146)

File: src/authorizations/components/CustomApiTokenOverlay.tsx (hunk 1 above)
Comment (blegesse-git) [reply src/authorizations/components/CustomApiTokenOverlay.tsx:43]: that makes sense. i've changed it to "Describe this new token"

File: src/authorizations/components/redesigned/CustomeApiTokenOverlay.test.tsx [hunk 5]
```diff
@@ -0,0 +1,29 @@
+import React from 'react'
```
Comment (hoorayimhelping) [inline src/authorizations/components/redesigned/CustomeApiTokenOverlay.test.tsx:1]: can you rename this file to `CustomApiTokenOverlay.test.tsx` (lose the `e` after `Custom`)

**Review by hoorayimhelping — APPROVED** [review]

File: src/authorizations/components/CustomApiTokenOverlay.tsx [hunk 6]
```diff
@@ -0,0 +1,54 @@
+import React, {FC, useState} from 'react'
```
Comment (Palakp41) [inline src/authorizations/components/CustomApiTokenOverlay.tsx:1]: We don't need this component at all. The only one we need to use is the one under `redesigned` folder.

File: src/authorizations/components/redesigned/CustomApiTokenOverlay.tsx [hunk 7]
```diff
@@ -1,7 +1,15 @@
-import React, {FC} from 'react'
+import React, {FC, useState} from 'react'
```
Comment (Palakp41) [inline src/authorizations/components/redesigned/CustomApiTokenOverlay.tsx:1]: This is the component that should render when `Custom Api Token`. I think something happened during rebase.

File: src/authorizations/components/redesigned/CustomApiTokenOverlay.tsx [hunk 8]
```diff
@@ -11,12 +19,36 @@ export const CustomApiTokenOverlay: FC<OwnProps> = props => {
...
+            alignItems={AlignItems.Center}
+            direction={FlexDirection.Column}
+            margin={ComponentSize.Large}
+          >
+            <Form.Element label="Description">
+              <Input
+                placeholder="Describe this new token"
+                value={description}
+                onChange={handleInputChange}
+                testID="custome-api-token-input"
```
Comment (Palakp41) [inline src/authorizations/components/redesigned/CustomApiTokenOverlay.tsx:46]: nit: Typo. `custom-api-token-input`.

File: src/authorizations/components/redesigned/CustomeApiTokenOverlay.test.tsx [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+import {render, fireEvent} from '@testing-library/react'
+import {CustomApiTokenOverlay} from '../CustomApiTokenOverlay'
+
+describe('CustomeApitokenDescription', () => {
```
Comment (Palakp41) [inline src/authorizations/components/redesigned/CustomeApiTokenOverlay.test.tsx:5]: nit: Typo. `CustomApitokenDescription`

**Review by Palakp41 — CHANGES_REQUESTED** [review]
Just a tiny change, delete the duplicate component file and this should be good to merge :)

**Review by Palakp41 — APPROVED** [review]

**Pull request merged**